### PR TITLE
implementation of velocity bounds for path vector.

### PR DESCRIPTION
**Pull request by olivier-roussel**

### CODE REVIEW

**Comment by hrp2-14** [comment]
Hi ! This project doesn't usually accept pull requests on master. If this wasn't intentionnal, you
can change the base branch of this pull request to devel (No need to close it for that). Best, a bot.

File: src/path-vector.cc [hunk 1]
```diff
@@ -146,6 +146,33 @@ namespace hpp {
...
+        int one = (reversed ? -1 : 1);
+        vector_t localbound (vector_t::Zero(bound.size()));
+        for (std::size_t i = iinit + one;
+            (reversed && i > iend) || i < iend;
+            i += one) {
+          paths_[i]->velocityBound(localbound, paths_[i]->timeRange().first, paths_[i]->timeRange().second);
+          bound = bound.cwiseMax(localbound);
+        }
+        paths_[iend]->velocityBound(localbound, Iinit(paths_[iend]->timeRange(), reversed), localtend);
+        bound = bound.cwiseMax(localbound);
```
Comment (jmirabel) [inline src/path-vector.cc:172]: Same here.
```suggestion
        bound = bound.cwiseMin(localbound);
```

Comment (florent-lamiraux) [reply src/path-vector.cc:172]: I would write `cwiseMax` as well. We want to get an upper bound on the velocities.

**Comment by florent-lamiraux** [comment]
Thank you Olivier. Would you be kind enough to add a unit test ?

File: src/path-vector.cc (hunk 1 above)
Comment (jmirabel) [reply src/path-vector.cc:172]: Indeed, my bad.

**Comment by olivier-roussel** [comment]
Sure I will try do that, but probably not before next week. Thanks both of you for your feedback. I still have an assertion in continuous collision (line 167 of `body-pair-colision.cc`, with `distanceLowerBound > 0` check failed), but this seems unrelated to this PR as I got same results with other kind of paths. This is strange as collision checking results looks OK.

**Comment by nim65s** [comment]
The CI is also showing an issue in this PR in debug mode, which is not triggered in the current devel:
```
 1/20 Test  #3: test-solid-solid-collision .........***Failed    0.10 sec
Running 1 test case...
lleg5_joint, lleg4_joint, lleg3_joint, lleg2_joint, lleg1_joint, rleg1_joint, rleg2_joint, rleg3_joint, rleg4_joint, rleg5_joint, 
test-solid-solid-collision: /opt/openrobots/include/pinocchio/spatial/symmetric3.hpp:418: pinocchio::Symmetric3Tpl<Scalar, Options> pinocchio::Symmetric3Tpl<Scalar, Options>::rotate(const Eigen::MatrixBase<OtherDerived>&) const [with D = Eigen::Matrix<double, 3, 3>; _Scalar = double; int _Options = 0]: Assertion `isUnitary(R.transpose()*R) && "R is not a Unitary matrix"' failed.
unknown location(0): fatal error: in "test_hpp_core/solid_solid_collision_1": signal: SIGABRT (application abort requested)
/root/robotpkg/path/hpp-core/work/hpp-core-4.11.0/tests/test-solid-solid-collision.cc(71): last checkpoint: "solid_solid_collision_1" test entry
*** 1 failure is detected in the test module "solid_solid_collision"
```

**Comment by olivier-roussel** [comment]
This is quite surprising as this test is not related to the modifications in this PR. I have checked and I got the same failure in the current devel. Looking briefly to it it looks like an uninitialized rotation matrix so I guess this might have passed the CI due to the undefined values and some bad luck. 
I think the `Transform3f` given in lines 85 and 88 of `test-solid-solid-collision.cc` were expected to be given as identity elements, i.e. we should have 
```
  int frame_id = robot->model().addFrame(::pinocchio::Frame("base_link",0,0,Transform3f::Identity(),::pinocchio::BODY));
  BOOST_CHECK(frame_id>=0);
  GeomIndex idObj = robot->geomModel().addGeometryObject
    (::pinocchio::GeometryObject("obstacle", frame_id, 0, box, Transform3f::Identity(), "", vector3_t::Ones()),
```
instead of 
```
  int frame_id = robot->model().addFrame(::pinocchio::Frame("base_link",0,0,Transform3f(),::pinocchio::BODY));
  BOOST_CHECK(frame_id>=0);
  GeomIndex idObj = robot->geomModel().addGeometryObject
    (::pinocchio::GeometryObject("obstacle", frame_id, 0, box, Transform3f(), "", vector3_t::Ones()),
```

**Comment by olivier-roussel** [comment]
If this is correct, I still can add these corrections to the PR (despite I believe they are out of its scope).

**Comment by florent-lamiraux** [comment]
Yes please, add a commit with these modifications.

**Comment by nim65s** [comment]
Thanks a lot for digging into this ! I couldn't reproduce the issue on the current devel, I really wonder where the difference is. Could you push that in this PR, so that we double-check if the CI is happier ? I agree this is out of scope, but the alternative is to make another PR with this fix, merge it, and then rebase the current PR on the updated devel. As you prefer :)

**Comment by nim65s** [comment]
Ok, now I understand !

This issue was already fixed in devel by #248, and I didn't check that your PR was based on the current devel, sorry for that :/
I hope I will catch this kind of things faster next time ^^

**Pull request merged**